Sources/Utility/Process.swift

```diff
@@ -165,6 +172,24 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Validate that the program provided exists on the search path and is executable.
+    internal func validateArgumentIsExecutable(_ program: String) throws {
```

Comment:
What would be preferred? I made it `internal` so that I could test it - when I started, I was making it `private`, but then even @testable import Utility couldn't resolve it to test it (a bit surprising to me philosophically - I'm used to testing private methods in other languages). I could just make it public, but it didn't seem like it necessarily fit in that scenario. Very open to feedback though - what do you think would make sense here?